### DISCUSSION TITLE
What is a handler to a view?

### DISCUSSION THREAD

**Discussion by mcguu** [dotnet/maui#5048] [Q&A] [answered]
what is the handler do?

how to understand a handler?

**Comment by hartez** [comment, marked as answer]
For each cross-platform View, there is a Handler which creates the real control on the target platform, maps all of the properties from the cross-platform View to the target control, and handles some other bookkeeping (wiring up events, etc.).

For example, a cross-platform MAUI Button has properties like `Text` and `FontSize`. MAUI has an iOS ButtonHandler which creates and manages a `UIButton` on the iOS platform; that ButtonHandler knows how to map the cross-platform `Text` property to `UIButton.Title`. The Windows ButtonHandler creates and manages a `Microsoft.UI.Xaml.Controls.Button`, and knows how to map `Text` to the `Content` property. 

You can use the built-in Handlers (and customize them), or you can create your own. If you're familiar with Renderers from Forms, you can think of Handlers as a more lightweight and flexible version of Renderers.

**Reply by mcguu** [reply]
Thank you @hartez  I wonder if there can donate to someone who answer question.

so the handlers is a real control to the view.

there is a `grahpicsView `and how to get events in the handler? 
If I want to customize a view which contains a `GrahpicsView `and event handler I should create a event handler and then a GraphicsView?

```
public class Interactions
{
    public Action StartInteraction{get;set;}
}

public class MyView:GraphicsView,Interactions ///?it is right?
{
    public MyView()
        {
            Drawable=new MyViewDrawable();
        }
}

```


then in platform/android/Interactions.cs to start the Interaction?
